src/main/java/appeng/container/guisync/GuiSync.java

```diff
@@ -19,11 +19,18 @@
 package appeng.container.guisync;
 
 
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
 
 
+/**
+ * Annotates that this field should be synchronized between the server and client.
+ * Requires the field to be public.
+ */
 @Retention( RetentionPolicy.RUNTIME )
+@Target( ElementType.FIELD )
 public @interface GuiSync
```

Comment:
Me too. Especially as it would make it easier to identify already used channels (assigned in superclasses).

Someone could argue that it should address them in a continuous way, but that is even currently true. But should be another issue/PR to fix it.